pkg/catalog/inbound_traffic_policies.go

```diff
@@ -195,10 +195,14 @@ func (mc *MeshCatalog) getRoutingRulesFromTrafficTarget(trafficTarget access.Tra
 	}
 
 	// Compute the allowed downstream service identities for the given TrafficTarget object
-	trustDomain := mc.GetTrustDomain()
+	trustDomains := mc.certManager.GetTrustDomains()
 	allowedDownstreamPrincipals := mapset.NewSet()
 	for _, source := range trafficTarget.Spec.Sources {
-		allowedDownstreamPrincipals.Add(trafficTargetIdentityToSvcAccount(source).AsPrincipal(trustDomain))
+		allowedDownstreamPrincipals.Add(trafficTargetIdentityToSvcAccount(source).AsPrincipal(trustDomains.Signing))
```

Comment:
nit: feels a bit more idiomatic to have it return a slice of trustdomains

Reply:
not sure I understand how it would be miss used but get the idiomatic point.  my thought was I didn't want to loop through a slice since we knew the values upfront.  I am likely going to modify this for #5160.  Maybe can try something different there?  Do you have any thoughts on how to looping through the slice?

Reply:
just a simple double loop, one over principles (sources in this line), and another over the slice of trust domains returned, ie:

```go
for _, source := range trafficTarget.Spec.Sources {
  for _, td := range trustDomains {
     allowedDownstreamPrincipals.Add(trafficTargetIdentityToSvcAccount(source).AsPrincipal(td))
  }
}
```

Reply:
hmm, since https://github.com/openservicemesh/osm/pull/5193#discussion_r993584627, I am leaning against this.  Would it be simpler if I got rid of trustdomian struct and just return the two strings and do a compare everywhere?